compiler/vendor/golang.org/x/tools/go/gcexportdata/gcexportdata.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package gcexportdata provides functions for locating, reading, and
+// writing export data files containing type information produced by the
+// gc compiler.  This package supports go1.7 export data format and all
+// later versions.
+//
+// Although it might seem convenient for this package to live alongside
+// go/types in the standard library, this would cause version skew
+// problems for developer tools that use it, since they must be able to
+// consume the outputs of the gc compiler both before and after a Go
+// update such as from Go 1.7 to Go 1.8.  Because this package lives in
+// golang.org/x/tools, sites can update their version of this repo some
+// time before the Go 1.8 release and rebuild and redeploy their
+// developer tools, which will then be able to consume both Go 1.7 and
+// Go 1.8 export data files, so they will work before and after the
+// Go update. (See discussion at https://github.com/golang/go/issues/15651.)
+//
+package gcexportdata // import "golang.org/x/tools/go/gcexportdata"
```

Comment:
Removing the now-unused old `gcimporter15` package from `vendor` directory is the right thing to do. However, we shouldn't vendor the new `gcexportdata` package.

This project currently follows the pattern of vendoring as few packages as possible, only the ones that _have_ to be vendored. The new `gcexportdata` package is not deprecated (nor removed from upstream) and doesn't need to be vendored at this time.

That means when newer versions of unvendored dependencies are released, installing the latest version of GopherJS will automatically use the latest versions of said dependencies. If the dependencies are vendored, they're not updated until we manually do so.

At this time, we prefer the reduced maintenance burden by not vendoring when it's not required. This PR should follow the current pattern and not deviate from it.

In the PR description, you said:

> we instead vendor golang.org/x/tools/go/gcexportdata and
> golang.org/x/tools/go/internal/gcimporter to avoid hassle for people
> upgrading to this version of GopherJS.

I don't understand what hassle you're referring to. To update to a newer version of GopherJS, per the [installation instructions](https://github.com/gopherjs/gopherjs#installation-and-usage), people do `go get -u github.com/gopherjs/gopherjs`. That automatically gets the latest versions of dependencies. Please let me know if I'm misunderstanding something, but I don't believe there is an increased hassle for users. (And if there is, we should deal with this outside of this PR if it affects other non-vendored packages too.)

So, please remove the newly vendored packages from this PR, just keep the deletion of `gcimporter15`.